lib/senkyoshi/models/file_resource.rb

```diff
@@ -0,0 +1,23 @@
+require "senkyoshi/models/resource"
+
+module Senkyoshi
+  ##
+  # Class to represent a resource constructed from a single 'dat' file.
+  ##
+  class FileResource < Resource
+    attr_reader(:id)
+
+    def initialize(id = nil)
+      @id = id
+    end
+
+    def self.from(xml = nil, pre_data = nil, _resource_xids = nil)
+      resource = new(pre_data[:file_name])
```

Comment:
Yeah it makes sense to remove `xml = nil`, `pre_data = nil`. I will do that. I'm not sure why I added that in the first place.